Address more Safer CPP warnings in WebCore/platform/cocoa and WebCore/platform/cg

cdumez opened the pull request:
#### d971a981f7bf150f46f0e8aab1043d6e280c2cbb
<pre>
Address more Safer CPP warnings in WebCore/platform/cocoa and WebCore/platform/cg
<a href="https://bugs.webkit.org/show_bug.cgi?id=299067">https://bugs.webkit.org/show_bug.cgi?id=299067</a>

Reviewed by Ryosuke Niwa.

* Source/WebCore/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UnretainedLocalVarsCheckerExpectations:
* Source/WebCore/platform/graphics/ShareableBitmap.h:
(WebCore::ShareableBitmapConfiguration::protectedPlatformColorSpace const):
* Source/WebCore/platform/graphics/cg/ShareableBitmapCG.mm:
(WebCore::ShareableBitmap::createFromImagePixels):
(WebCore::ShareableBitmap::createGraphicsContext):
(WebCore::ShareableBitmap::paint):
(WebCore::ShareableBitmap::createPlatformImage):
* Source/WebCore/platform/graphics/cocoa/GraphicsContextGLCocoa.mm:
(WebCore::GraphicsContextGLCocoa::surfaceBufferToVideoFrame):
(WebCore::GraphicsContextGLCocoa::insertFinishedSignalOrInvoke):
* Source/WebCore/platform/graphics/cocoa/HEVCUtilitiesCocoa.mm:
(WebCore::validateHEVCParameters):
(WebCore::parseStringArrayFromDictionaryToUInt16Vector):
* Source/WebCore/platform/graphics/cocoa/IOSurface.h:
* Source/WebCore/platform/graphics/cocoa/IOSurface.mm:
(WebCore::surfaceNameToNSString):
(WebCore::IOSurface::create):
(WebCore::createSurfaceViaCoreVideo):
(WebCore::optionsForBiplanarSurface):
(WebCore::optionsForSurface):
(WebCore::createSurface):
(WebCore::IOSurface::createCompatibleBitmap):
(WebCore::IOSurface::createPlatformContext):
(WebCore::IOSurface::createBitmapPlatformContext):
(WebCore::IOSurface::setColorSpaceProperty):
* Source/WebCore/platform/graphics/cocoa/IconCocoa.mm:
(WebCore::Icon::create):
(WebCore::Icon::paint):
* Source/WebCore/platform/graphics/cocoa/WebCoreDecompressionSession.mm:
(WebCore::WebCoreDecompressionSession::assignResourceOwner):
(WebCore::WebCoreDecompressionSession::ensureDecompressionSessionForSample):
(WebCore::copyDescriptionExtensionValuesIfNeeded):
(WebCore::WebCoreDecompressionSession::decodeSampleInternal):
* Source/WebCore/platform/graphics/cocoa/WebProcessGraphicsContextGLCocoa.mm:

Canonical link: <a href="https://commits.webkit.org/300202@main">https://commits.webkit.org/300202@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/712aeb5ab848d9e88882d9bd67fbce8b0bf963c3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/121529 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/41226 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/31885 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/127987 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/73612 "Built successfully") | [❌ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b10d07d9-57f5-4942-b230-c33b1192a5c3) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/123405 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/41928 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/49805 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/92302 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61404 "Passed tests") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b4c0730f-1e13-49c6-94a3-808edab00571) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/124481 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33480 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/108859 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/72979 "Found 1 new API test failure: WPE/TestWebKitWebContext:/webkit/WebKitWebContext/no-web-process-leak (failure)") | | [❌ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/bbba4b6e-e601-48c3-8929-8eba1f861f93) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32488 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/27026 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/71555 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/102963 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/27199 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/130804 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/48457 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/36848 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/100893 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/48825 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/105084 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/100799 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25589 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46213 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/24294 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/45153 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/48316 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/54028 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/47787 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/51133 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/49469 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/352ed8e75211deb66d00da6a0b875c5059bc3b97. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50915

cdumez committed:
Address more Safer CPP warnings in WebCore/platform/cocoa and WebCore/platform/cg
https://bugs.webkit.org/show_bug.cgi?id=299067

Reviewed by Ryosuke Niwa.

* Source/WebCore/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UnretainedLocalVarsCheckerExpectations:
* Source/WebCore/platform/graphics/ShareableBitmap.h:
(WebCore::ShareableBitmapConfiguration::protectedPlatformColorSpace const):
* Source/WebCore/platform/graphics/cg/ShareableBitmapCG.mm:
(WebCore::ShareableBitmap::createFromImagePixels):
(WebCore::ShareableBitmap::createGraphicsContext):
(WebCore::ShareableBitmap::paint):
(WebCore::ShareableBitmap::createPlatformImage):
* Source/WebCore/platform/graphics/cocoa/GraphicsContextGLCocoa.mm:
(WebCore::GraphicsContextGLCocoa::surfaceBufferToVideoFrame):
(WebCore::GraphicsContextGLCocoa::insertFinishedSignalOrInvoke):
* Source/WebCore/platform/graphics/cocoa/HEVCUtilitiesCocoa.mm:
(WebCore::validateHEVCParameters):
(WebCore::parseStringArrayFromDictionaryToUInt16Vector):
* Source/WebCore/platform/graphics/cocoa/IOSurface.h:
* Source/WebCore/platform/graphics/cocoa/IOSurface.mm:
(WebCore::surfaceNameToNSString):
(WebCore::IOSurface::create):
(WebCore::createSurfaceViaCoreVideo):
(WebCore::optionsForBiplanarSurface):
(WebCore::optionsForSurface):
(WebCore::createSurface):
(WebCore::IOSurface::createCompatibleBitmap):
(WebCore::IOSurface::createPlatformContext):
(WebCore::IOSurface::createBitmapPlatformContext):
(WebCore::IOSurface::setColorSpaceProperty):
* Source/WebCore/platform/graphics/cocoa/IconCocoa.mm:
(WebCore::Icon::create):
(WebCore::Icon::paint):
* Source/WebCore/platform/graphics/cocoa/WebCoreDecompressionSession.mm:
(WebCore::WebCoreDecompressionSession::assignResourceOwner):
(WebCore::WebCoreDecompressionSession::ensureDecompressionSessionForSample):
(WebCore::copyDescriptionExtensionValuesIfNeeded):
(WebCore::WebCoreDecompressionSession::decodeSampleInternal):
* Source/WebCore/platform/graphics/cocoa/WebProcessGraphicsContextGLCocoa.mm:

Canonical link: https://commits.webkit.org/300202@main

webkit-commit-queue commented:
Committed 300202@main (d971a981f7bf): <https://commits.webkit.org/300202@main>

Reviewed commits have been landed. Closing PR #50915 and removing active labels.
